trin-core/src/types/accumulator.rs

```diff
@@ -123,7 +134,7 @@ impl MasterAccumulator {
         };
 
         // check if the epoch accumulator is full
-        if self.current_epoch.header_records.len() == EPOCH_SIZE {
+        if self.next_header_height() > 0 && self.next_header_height() % EPOCH_SIZE as u64 == 0 {
```

Comment:
Why do we need the more complicated condition?

Reply:
Can't remember exactly which one, but there was an edge case that wasn't being handled properly in the `validation_data_matches_valid()` test

Reply:
Which test was failing? Looking at it more closely, this logic is incorrect.

- Assume `EPOCH_SIZE` is equal to `8192`.
- Suppose `next_header_height` is equal to `8192 * 2` such that we are about to fill the current (second) epoch.
- The condition returns `true` because `8192 * 2 > 0` and `(8192 * 2) % 8192` is equal to `0`.

Reply:
Good catch, you're right that the extra logic was unnecessary. But I believe the extra logic was correct. If we are trying to update the accumulator with block #`16384` (8192* 2) (i.e. `next_header_height == 16384`), this means that we already have `16384` total header records (blocks 0 -16383). In which case our current epoch accumulator is full and we want to clear it, update our historical headers and then push block #`16384` as the first record in the new current epoch. Regardless, that extra logic is redundant and I've reverted it back to the simpler check.

Reply:
I forgot that height starts at zero 🤦‍♂️